pkg/util/containers/metrics/containerd/collector.go

```diff
@@ -75,10 +76,17 @@ func newContainerdCollector(cache *provider.Cache) (provider.CollectorMetadata,
 		ContainerIDForPID: provider.MakeRef[provider.ContainerIDForPIDRetriever](collector, collectorPriority),
 	}
 
+	kataCollectors := &provider.Collectors{
+		Stats:             provider.MakeRef[provider.ContainerStatsGetter](collector, collectorPriority),
+		PIDs:              provider.MakeRef[provider.ContainerPIDsGetter](collector, collectorPriority),
+		ContainerIDForPID: provider.MakeRef[provider.ContainerIDForPIDRetriever](collector, collectorPriority),
```

Comment:
This would not work with Kata containers right?

Reply:
When we inspect a pod, a pid is reported. I think it corresponds to the pid of the virtual machine. So technically it still gives a mapping from pid to ContainerID.

But it is true that we most likely won't have to use it given that dogstatsd is the only caller and it requires to mount the UDS socket, which is not supported by kata containers anyway

Reply:
I think you can only register `Stats`